tests/sorting_products.js

```diff
@@ -1,29 +1,76 @@
-import { Selector, t, ClientFunction }  from "testcafe";
-
-import { loginButton, validUserName, validPassword, 
-        userNameInput, passwordInput,
-        productSortDrop, sortProductAZ, productTitle } from "./login.js";
+import loginPage from "../Pages/loginPage.js";
+import productsPage from "../Pages/productsPage.js";
+const validUserName = 'standard_user';
+const validPassword = 'secret_sauce';
 const baseUrl = 'https://www.saucedemo.com/';
 
+
 fixture`Verify sorting options`
     .page(baseUrl)
 
+  //MAKES SENSE: (A TO Z, Z to A, si preturi) sort simplu, click pe optin, luat lista, comparat liste.
+    
+test(`Sort products by Name (A to Z)`, async t => { 
+    await loginPage.typeUserName(validUserName);
+    await loginPage.typePassword(validPassword);
+    await loginPage.clickOnLoginBttn();
+  
+    const productTitleElements = await productsPage.productTitle;
+    const productTitles = [];
+  
+    for (let i = 0; i < await productTitleElements.count; i++) {
+      const title = await productTitleElements.nth(i).innerText;
+      console.log(`Title ${i}: ${title}`);
+      productTitles.push(title);
+    }
+  
+    console.log('Product titles: ', productTitles);
 
-test.skip(`Sort products by Name (A to Z)`, async t => {
-    await t
-        .typeText(userNameInput, validUserName)
-        .typeText(passwordInput, validPassword)
-        .click(loginButton)
-        .click(productSortDrop)
-        .click(sortProductAZ)
+    await productsPage.openSortDrop();
+    await productsPage.sortByDescending();
+    await productsPage.openSortDrop();
+    await productsPage.sortByAscending();
+    
+    const sortedProductTitleElements = await productsPage.productTitle;
+    const sortedTitles = [];
+    
+    for (let i = 0; i < await sortedProductTitleElements.count; i++) {
+      const title = await sortedProductTitleElements.nth(i).innerText;
+      console.log(`Sorted Title ${i}: ${title}`);
+      sortedTitles.push(title);
+    }
+    
+    await t.expect(productTitles).eql(sortedTitles);
```

Comment:
productTitles needs to be sorted before this assertion. Right now you're just comparing the initial list, which you're assuming to be sorted, but the test logic is not 100% correct this way. This may work now, but we want the test to be written in a flexible manner, so that regardless of the initial list, the test will run correctly.